client/src/components/App.js

```diff
@@ -33,7 +33,7 @@ const App = (props) => {
         <Route exact path="/users/new" component={RegistrationForm} />
         <Route exact path="/user-sessions/new" component={SignInForm} />
         <Route exact path="/dogs" component={DogsList} />
-        <Route exact path="/dogs/:id" component={DogShowPage} />
+        <Route exact path="/dogs/:id" render={(props) => <DogShowPage user={currentUser} {...props} />} />
```

Comment:
If you are trying to pass the `currentUser` and still have access to things like `props.match`, we should instead:
```js